utils/plot-utils.ipynb

```diff
@@ -29,14 +29,24 @@
     "    .register((figure, ctx) -> {\n",
     "        ctx.renderIfRequested(io.github.spencerpark.jupyter.kernel.display.mime.MIMEType.TEXT_HTML, () -> {\n",
     "            String id = UUID.randomUUID().toString().replace(\"-\", \"\");\n",
+    "            // Generate Image instead of JS\n",
     "            boolean plotImage = System.getenv().containsKey(\"D2L_PLOT_IMAGE\");\n",
+    "            /* Plotly preloaded to website\n",
+    "             * By default, require.js is required to render js properly and it has compatibility issue with\n",
+    "             * jquery and other js packages. This module is typically used in NodeJS.\n",
+    "             */\n",
+    "            boolean preloadPlotly = System.getenv().containsKey(\"PRELOAD_PLOTLY\");\n",
     "\n",
     "            figure.asJavascript(id);\n",
     "            Map<String, Object> context = figure.getContext();\n",
     "            StringBuilder html = new StringBuilder();\n",
     "            html.append(\"<img id=\\\"\").append(id).append(\"_img\\\"></img>\\n\");\n",
     "            html.append(\"<div id=\\\"\").append(id).append(\"\\\"></div>\\n\");\n",
-    "            html.append(\"<script>require(['https://cdn.plot.ly/plotly-1.57.0.min.js'], Plotly => {\\n\");\n",
+    "            if (!preloadPlotly) {\n",
+    "                html.append(\"<script>require(['https://cdn.plot.ly/plotly-1.57.0.min.js'], Plotly => {\\n\");\n",
```

Comment:
If we already pre-loaded, should we just include the js from local?

Reply:
What do you mean by include js from local? On the jupyter notebook side, plot.ly is not loaded. You have to use node.js require feature to dynamic load the js in the runtime.